### ISSUE TITLE
Plugin design: Do inference to visualize network activations

### ISSUE THREAD

**Issue by JarnoRFB** [tensorflow/tensorboard#527]
Is it conceivable that one could write a [plugin](https://github.com/tensorflow/tensorboard-plugin-example) that visualizes activations similar to the tool from [Yosinski](https://github.com/yosinski/deep-visualization-toolbox)? I wonder, because it seems that one should rather write summaries, that are then processed by the backend, but if one wants to interactively choose inputs, that does not seem appropriate. Would it make sense to just configure the backend with the `graph_def` and a path to a checkpoint and then serve inferred activations? I am asking, because I currently work on a tool simliar to Yosinski's and integration into Tensorboard would be nice, but I am not sure if this complies with the intended design.

**Comment by chihuahua** [comment]
@JarnoRFB, thank you for bringing Yosinski's tool to our attention! I would love to explore its scope more.

Interactively choosing inputs and running them through conv filters does seem like something summaries today don't support. Summaries just log data during model runs.

However, the solution you proposed - performing inference on a model loaded from a checkpoint does seem like it would work. We could introduce a plugin specific flag for the checkpoint, and the plugin could use `tf.train.Saver` to load the checkpoint.

My hunch is ... I would love to try this out. And I think some teams at Google could benefit from this.

@jart @wchargin @dandelionmane

**Comment by chihuahua** [comment]
All in all, I believe this is doable. We have a means of communicating with the model (inference using the checkpoint), and we have a means of serving data to the client.

**Comment by wchargin** [comment]
Duplicate of #81?

**Comment by chihuahua** [comment]
Yeah, it seems like a dup of #81. @cesarsalgado

I guess this issue has somebody who wants to take action though.

**Comment by JarnoRFB** [comment]
Thanks for your replies. The feature request is definitely a duplicate of #81, but the idea of doing inference in the backend of tensorboard has not been explored so far, as e.g. https://github.com/InFoCusp/tf_cnnvis just writes image summaries, making it hard to choose the image you want to visualize for from tensorboard. I would love to try to built a prototype for this, if I can carve out some time. Happy about any help or suggestions.

**Comment by chihuahua** [comment]
Yes - come be our pioneer. :) To be clear though, the project sounds very much tenable.

**Comment by bileschi** [comment]
@caisq is there more that could be said here?  Is there a master issue for looking at values?
